### ISSUE TITLE
Attach Gradle sources when "all" distribution is used 

### ISSUE THREAD

**Issue by donat** [eclipse/buildship#330]
Original post: https://bugs.eclipse.org/bugs/show_bug.cgi?id=509079

**Comment by adammurdoch** [comment]
We should fix this inside Gradle, so that the library definition in the Eclipse model points to the correct source.

**Comment by oehme** [comment]
Yes, this is easy to do in the `classpath.file.whenMerged()` hook.

**Comment by dwaal** [comment]
Not so easy after all - issue still open?

Many devs I know including myself find themselves in the situation of spending a big chunk of their time fiddling around with Gradle. Now it seeps more and more into IDEs and makes them buggy and slow.

Maybe this is just a painful period before it all gets better but my understanding is, that the design of Gradle in general (on many levels, starting with Groovy - I guess Javascript was not available) and the deep marriage with IDEs is just a bad idea.

It's just one big black box you cannot escape anymore. In 2018 you are forced to use Gradle.

**Comment by dwaal** [comment]
It's not a rant. Gradle is the new regex.

**Comment by dwaal** [comment]
No point in opening issues. The very foundation of Gradle is the problem. Buildship is bad by extension.

But it doesn't matter. The fix is in and Gradle is inescapable if you do anything in the extended Java universe. My personal solution is to wait until Stockholm-syndrome hopefully soon kicks in.

**Comment by wordlessly** [comment]
We have similar code running. We wasted a lot of time on this and other Gradle/Groovy hacks but the issue goes back to 2015 and at some point we needed to fix it.
